terraform/modules/dandiset_bucket/main.tf

```diff
@@ -0,0 +1,79 @@
+resource "aws_s3_bucket" "dandiset_bucket" {
+  # provider = var.aws_provider
+
+  bucket = var.bucket_name
+  // Public access is granted via a bucket policy, not a canned ACL
+  acl = "private"
+
+  cors_rule {
+    allowed_origins = [
+      "*",
+    ]
+    allowed_methods = [
+      "PUT",
+      "POST",
+      "GET",
+      "DELETE",
+    ]
+    allowed_headers = [
+      "*"
+    ]
+    expose_headers = [
+      "ETag",
+    ]
+    max_age_seconds = 3000
+  }
+
+  logging {
+    target_bucket = aws_s3_bucket.log_bucket.id
+  }
+
+  versioning {
+    enabled = var.versioning
+  }
+
+  lifecycle {
+    prevent_destroy = true
```

Comment:
Do you want this for the staging bucket too?

Reply:
I think it's a good idea, yeah. Users will probably expect test data to stick around in staging, so we should be deliberate if we ever delete the entire bucket.